src/app/server/Server.h

```diff
@@ -307,6 +314,22 @@ struct CommonCaseDeviceServerInitParams : public ServerInitParams
         }
 #endif
 
+#if CHIP_CONFIG_TC_REQUIRED
+        static app::DefaultTermsAndConditionsProvider sDefaultTermsAndConditionsProviderInstance;
+
+        if (this->termsAndConditionsProvider == nullptr)
+        {
+            Optional<app::TermsAndConditions> termsAndConditions = Optional<app::TermsAndConditions>({
+                .value   = CHIP_CONFIG_TC_REQUIRED_ACKNOWLEDGEMENTS,
+                .version = CHIP_CONFIG_TC_REQUIRED_ACKNOWLEDGEMENTS_VERSION,
+            });
+
+            ReturnErrorOnFailure(
+                sDefaultTermsAndConditionsProviderInstance.Init(this->persistentStorageDelegate, termsAndConditions));
+            this->termsAndConditionsProvider = &sDefaultTermsAndConditionsProviderInstance;
+        }
+#endif
```

Comment:
Do not use build time configs for this. The T&C provider needs to always be injected if needed. That would avoid baking-in complex fallback rules for build-time versus dynamic.